solutions/cses-1091.mdx

```diff
@@ -51,3 +55,90 @@ int main(){
 	return 0;
 }
 ```
+
+</CPPSection>
+
+<JavaSection>
+## Implementation
+
+[Unofficial Editorial (C++)](https://codeforces.com/blog/entry/83295)
+
+**Time Complexity:** $\mathcal{O}(n+ m)$
+
+For each customer, we want to find the maximum possible ticket price they will
+accept from the available tickets. If the customer's maximum price is too low,
+so that no ticket price is below it, we can return -1. If not, we find the
+greatest ticket price less than or equal to the maximum price, and then remove
+that price as specified by the problem.
+
+We use, at least in theory, a multiset for this. A
+[multiset](https://en.wikipedia.org/wiki/Multiset) is basically an ordered set
+that allows multiple pairs of the same key. But Java doesn't have a defined
+multiset method, so we have to use a regular ordered set with some special
+modifications, using values as a counter of the number of identical sets. This
+allows us to distinguish between sets, and remove each set once.
+
+<Warning>
+Due to the way CSES's grader works (allowing 1 second for all problems + being slow), test cases 5 and 8 TLE (time limit exceeded). USACO gives Java test cases 4 seconds, so this isn't representative of what competing with Java will be like. 
+</Warning>
+```java
+import java.io.*;
+import java.util.*;
+
+public class ConcertTickets {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+
+		int ticketNum = io.nextInt();
+		int peopleNum = io.nextInt();
+		// No multiset in java so we'll have to use a TreeMap
+		TreeMap<Integer, Integer> ticketMultiset = new TreeMap<>();
+		Map.Entry<Integer, Integer> val;
+
+		for (int i = 0; i < ticketNum; i++) {
+			int priceOfTicket = io.nextInt();
+			/*
+			 * If the set already contains the same key value, just create another pair 
+			 * right next to it
+			 */
+			if (ticketMultiset.containsKey(priceOfTicket)) {
+				ticketMultiset.put(priceOfTicket, ticketMultiset.get(priceOfTicket) + 1);
+			} else {
+				ticketMultiset.put(priceOfTicket, 1);
+			}
+		}
+
+		for (int i = 0; i < peopleNum; i++) {
+			int customerMaxPrice = io.nextInt();
+			/*
+			 * Get the optimal ticket for this customer 
+			 * We add 1 because lowerEntry returns the highest *strictly* lower entry
+			 */
+			val = ticketMultiset.lowerEntry(customerMaxPrice + 1); 
+			/*
+			 * If there is no lower price, the answer will be null, 
+			 * so there's no possible prices, so we can exit and return -1
```

Comment:
```suggestion
			 * If there's no possible prices, we can exit and return -1
```

Reply:
if this makes the comment one line, just use an inline comment i mean